### PR TITLE
clarify some error messages

### PR DESCRIPTION
**Pull request by frasertweedale**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Some error messages related to userns mappings do not quite match
CRI-O's behaviour.  Furthermore, the same error message was used for
two different conditions.  Make the error messages clearer and more
actionable.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
Improved some error messages related to user namespaces.
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/ok-to-test
LGTM

File: server/sandbox_run_linux.go
Comment (TomSweeneyRedHat) [inline server/sandbox_run_linux.go:97]: I'll defer to @haircommander or @saschagrunert if this is right technically, but I think you want:
```suggestion
			return nil, errors.New("cannot use uidmapping or gidmapping if RunAsUser is not set")
```

File: server/sandbox_run_linux.go
Comment (TomSweeneyRedHat) [inline server/sandbox_run_linux.go:101]:
```suggestion
				return nil, errors.New("cannot use uidmapping or gidmapping if not running as root and minimum mappable ID is not set")
```

File: server/sandbox_run_linux.go
Comment (frasertweedale) [reply server/sandbox_run_linux.go:97]: Thanks for point it out.  I'll push an update on Monday.

/hold

**Comment by haircommander** [comment]
/retest-required
/lgtm

thanks!

**Comment by haircommander** [comment]
```
not ok 94 additional devices permissions
# (in test file ./devices.bats, line 61)
#   `[[ "$output" == *"Operation not permitted"* ]]' failed
```
these [crun failures](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_integration_crun_fedora/17117/s3/) seem to be common and unrelated to this cc @giuseppe

**Pull request merged**
